grav/hub.go

```diff
@@ -16,64 +17,49 @@ type hub struct {
 	nodeUUID    string
 	belongsTo   string
 	interests   []string
-	transport   Transport
+	mesh        MeshTransport
+	bridge      BridgeTransport
 	discovery   Discovery
-	context     context.Context
-	cancelFunc  func()
 	log         *vlog.Logger
 	pod         *Pod
 	connectFunc func() *Pod
 
-	connections      map[string]*connectionHolder
-	topicConnections map[string]TopicConnection
+	meshConnections   map[string]*connectionHandler
+	bridgeConnections map[string]BridgeConnection
 
 	capabilityUUIDBuffers map[string]*MsgBuffer
 
 	lock sync.RWMutex
 }
 
-type connectionHolder struct {
-	Conn      Connection
-	Signaler  *WithdrawSignaler
-	BelongsTo string
-	Interests []string
-}
-
 func initHub(nodeUUID string, options *Options, connectFunc func() *Pod) *hub {
-	ctx, cancel := context.WithCancel(context.Background())
-
 	h := &hub{
 		nodeUUID:              nodeUUID,
 		belongsTo:             options.BelongsTo,
 		interests:             options.Interests,
-		transport:             options.Transport,
+		mesh:                  options.MeshTransport,
+		bridge:                options.BridgeTransport,
 		discovery:             options.Discovery,
-		context:               ctx,
-		cancelFunc:            cancel,
 		log:                   options.Logger,
-		pod:                   connectFunc(),
 		connectFunc:           connectFunc,
-		connections:           map[string]*connectionHolder{},
-		topicConnections:      map[string]TopicConnection{},
+		meshConnections:       map[string]*connectionHandler{},
+		bridgeConnections:     map[string]BridgeConnection{},
 		capabilityUUIDBuffers: map[string]*MsgBuffer{},
 		lock:                  sync.RWMutex{},
 	}
 
-	// start transport, then discovery if each have been configured (can have transport but no discovery)
-	if h.transport != nil {
-		transportOpts := &TransportOpts{
+	// start mesh transport, then discovery if each have been configured (can have transport but no discovery)
+	if h.mesh != nil {
+		transportOpts := &MeshOptions{
 			NodeUUID: nodeUUID,
 			Port:     options.Port,
 			URI:      options.URI,
 			Logger:   options.Logger,
 		}
 
-		// setup messages to be sent to all active connections
-		h.pod.On(h.outgoingMessageHandler())
-
 		go func() {
-			if err := h.transport.Setup(transportOpts, h.handleIncomingConnection, h.findConnection); err != nil {
-				h.log.Error(errors.Wrap(err, "failed to Setup transport"))
+			if err := h.mesh.Setup(transportOpts, h.handleIncomingConnection); err != nil {
```

Comment:
Should we propagating transport setup errors so the caller can decide how it wants to handle the situation?